Assets/MixedRealityToolkit/Utilities/Editor/DependencyWindow.cs

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+
+using Object = UnityEngine.Object;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    public class DependencyWindow : EditorWindow
+    {
+        private Object assetSelection = null;
+        private int maxDisplayDepth = 8;
+        private float assetGraphRefreshTime = 0.0f;
+        private Vector2 scrollPosition = Vector2.zero;
+
+        private class AssetGraphNode
+        {
+            public string guid = null;
+            public HashSet<AssetGraphNode> assetsThisDependsOn = new HashSet<AssetGraphNode>();
+            public HashSet<AssetGraphNode> assetsDependentOnThis = new HashSet<AssetGraphNode>();
+        }
+
+        private Dictionary<string, AssetGraphNode> dependencyGraph = new Dictionary<string, AssetGraphNode>();
+
+        private const string windowTitle = "Dependency Window";
+        private const string guidPrefix = "guid: ";
+        private const string nullGuid = "00000000000000000000000000000000";
+        private const int guidCharacterCount = 32;
+
+        private readonly string[] assetsWithDependencies =
+        {
+            ".unity",
+            ".prefab",
+            ".asset",
+            ".mat",
+            ".anim",
+            ".controller"
+        };
+
+        [MenuItem("Mixed Reality Toolkit/Utilities/Dependency Window", false, 3)]
+        private static void ShowWindow()
+        {
+            var window = GetWindow<DependencyWindow>();
+            window.titleContent = new GUIContent(windowTitle, EditorGUIUtility.IconContent("d_EditCollider").image);
+            window.minSize = new Vector2(520.0f, 380.0f);
+            window.RefreshAssetGraph();
+            window.Show();
+        }
+
+        private void OnGUI()
+        {
+            MixedRealityEditorUtility.RenderMixedRealityToolkitLogo();
+
+            EditorGUILayout.LabelField("Mixed Reality Toolkit Dependency Window", new GUIStyle() {fontSize = 12, fontStyle = FontStyle.Bold });
+            EditorGUILayout.LabelField("This tool displays how assets reference and depend on each other. Dependencies are calculated by parsing guids within project YAML files, code dependencies are not considered.", EditorStyles.wordWrappedLabel);
+
+            EditorGUILayout.Space();
+
+            if (EditorSettings.serializationMode != SerializationMode.ForceText)
+            {
+                EditorGUILayout.HelpBox("Dependencies can only be tracked with text assets. Please change the project serialization mode to \"Force Text\" via Edit->Project Settings->Editor->Asset Serialization", MessageType.Error);
```

Comment:
Can you add a button to just launch the menu directly? I think this should work:
https://docs.unity3d.com/ScriptReference/EditorApplication.ExecuteMenuItem.html

Reply:
I'm not sure I can with a project setting? But, I just added a "Fix Now" buttoon like you find in the material inspector.